src/tz_android.rs

```diff
@@ -1,12 +1,16 @@
+use std::ffi::CStr;
+
 use android_system_properties::AndroidSystemProperties;
+use once_cell::sync::OnceCell;
 
 // From https://android.googlesource.com/platform/ndk/+/android-4.2.2_r1.2/docs/system/libc/OVERVIEW.html
 // The system property named 'persist.sys.timezone' contains the name of the current timezone.
 
-const TIMEZONE_PROP_KEY: &str = "persist.sys.timezone";
+static PROPERTIES: OnceCell<AndroidSystemProperties> = OnceCell::new();
 
 pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
-    AndroidSystemProperties::new()
-        .get(TIMEZONE_PROP_KEY)
+    PROPERTIES
+        .get_or_init(AndroidSystemProperties::new)
+        .get_from_cstr(unsafe { CStr::from_bytes_with_nul_unchecked(b"persist.sys.timezone\0") })
```

Comment:
Did the API change in the android crate that requires the unsafe here?

Reply:
I introduced the new method in https://github.com/nical/android_system_properties/pull/9. `AndroidSystemProperties::get()` still exists, and the new `AndroidSystemProperties::get_from_cstr()` is safe, too. The new function does not alloc a CString at runtime to get a properly null terminated `&CStr`.

[`CStr::from_bytes_with_nul_unchecked()`](https://doc.rust-lang.org/stable/core/ffi/struct.CStr.html#method.from_bytes_with_nul_unchecked) is unsafe, because you have to ensure that it ends with a `\0`, and there are no `\0` inside the string except for the terminator. I could have used the save method [`CStr::from_bytes_with_nul()`](https://doc.rust-lang.org/stable/core/ffi/struct.CStr.html#method.from_bytes_with_nul), but well. I trust my eyes that I see a `\0` at the end of the string. :)